options-manager.js

```diff
@@ -102,6 +98,14 @@ function buildConfig(opts) {
 		config.rules['babel/object-curly-spacing'] = [2, 'never'];
 	}
 
+	if (opts.rules) {
+		for (var attr in opts.rules) {
+			if (opts.rules.hasOwnProperty(attr)) {
+				config.rules[attr] = opts.rules[attr];
+			}
```

Comment:
You can just use the objectAssign ponyfill. We already depend on it https://github.com/sindresorhus/xo/blob/4b44968e18666bf22f742de9af12f4b61315cd8e/options-manager.js#L7